src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -109,11 +119,10 @@ public bool SnippetShortcutExists_NonBlocking(string shortcut)
         public virtual bool ShouldFormatSnippet(SnippetInfo snippetInfo)
             => false;
 
-        private void PopulateSnippetCaches()
+        private void PopulateSnippetCaches(IAsyncToken token)
```

Comment:
💭 It looks like this can be simplified further by making this method asynchronous. It's behaving asynchronously already, but hiding that fact behind a synchronous API.